atomic_reactor/cachito_util.py

```diff
@@ -0,0 +1,163 @@
+    """Invalid request made to Cachito's API"""
+
+
+class CachitoAPIUnsuccessfulRequest(CachitoAPIError):
+    """Cachito's API request not completed successfully"""
+
+
+class CachitpAPIRequestTimeout(CachitoAPIError):
+    """A request to Cachito's API took too long to complete"""
+
+
+class CachitoAPI(object):
+
+    def __init__(self, api_url, insecure=False, token=None, cert=None):
+        self.api_url = api_url
+        self.session = self._make_session(insecure=insecure, cert=cert)
+
+    def _make_session(self, insecure, cert):
+        # method_whitelist=False allows retrying non-idempotent methods like POST
+        session = get_retrying_requests_session(method_whitelist=False)
+        session.verify = not insecure
+        if cert:
+            session.cert = cert
+        return session
+
+    def request_sources(self, repo, ref, flags=None, pkg_managers=None, user=None):
+        """Start a new Cachito request
+
+        :param repo: str, the URL to the SCM repository
+        :param ref: str, the SCM reference to fetch
+        :param pkg_managers: list<str>, list of package managers to be used for resolving
+                             dependencies
+        :param flags: list<str>, list of flag names
+
+        :return: dict, representation of the created Cachito request
+        :raise CachitoAPIInvalidRequest: if Cachito determines the request is invalid
+        """
+        payload = {
+            'repo': repo,
+            'ref': ref,
+            'flags': flags,
+            'pkg_managers': pkg_managers,
+            'user': user,
+        }
+        # Remove None values
+        payload = {k: v for k, v in payload.items() if v}
+
+        url = '{}/api/v1/requests'.format(self.api_url)
+        logger.debug('Making request %s with payload:\n%s', url, json.dumps(payload, indent=4))
+        response = self.session.post(url, json=payload)
+        if response.status_code == 400:
+            raise CachitoAPIInvalidRequest(response.json()['error'])
+        response.raise_for_status()
+        return response.json()
+
+    def wait_for_request(
+            self, request_id, burst_retry=1, burst_length=30, slow_retry=10, timeout=3600):
+        """Wait for a Cachito request to complete
+
+        :param request_id: int, the Cachito request ID
+        :param burst_retry: int, seconds to wait between retries prior to exceeding
+                            the burst length
+        :param burst_length: int, seconds to switch to slower retry period
+        :param slow_retry: int, seconds to wait between retries after exceeding
+                           the burst length
+        :param timeout: int, when to give up waiting for compose request
+
+        :return: dict, latest representation of the Cachito request
+        :raise CachitoAPIUnsuccessfulRequest: if the request completes unsuccessfully
+        :raise CachitpAPIRequestTimeout: if the request does not complete timely
```

Comment:
```suggestion
        :raise CachitoAPIRequestTimeout: if the request does not complete in a timely manner
```

Reply:
Thank for catching this! :man_facepalming: